### ISSUE TITLE
Allow environment variable settings to be inherited or copied across projects

### ISSUE THREAD

**Issue by devstuff** [travis-ci/travis-ci#7128]
In our organization, our IT/Ops team own the Travis-CI secrets for accessing private resources. For a new repository, this currently requires us to send a ticket to IT/Ops and wait...

For self-service it would be a great idea to allow the variables to be inherited from the organization level, or be able to be copied from one project to another (within the same organization, and with the same visibility).

**Comment by jfhamlin** [comment]
Thank you for opening this (it's essentially a request for the same feature as https://github.com/travis-ci/travis-ci/issues/2069). It looks like CircleCI supports this with [Contexts](https://circleci.com/docs/2.0/contexts/). This is an important feature with widespread utility and clear security benefits; I hope TravisCI considers it soon.

cc @BanzaiMan

**Comment by andye2004** [comment]
Hi @BanzaiMan, looks like the initial request for this was opened over a year ago, is there any indication of whether this will actually be implemented, if so when?

TBH I'm evaluating Travis alongside other CI tools and the lack of this feature will likely prove decisive.

Thanks, Andy.

**Comment by BanzaiMan** [comment]
I am sorry to inform you that there has been no update on this issue.

**Comment by destinyb-realityi** [comment]
@BanzaiMan is there any particular reason this hasn't been implemented? From research this has been requested for around 3 to 4 years.

**Comment by andye2004** [comment]
@BanzaiMan The reason I was prodding you was because I had opened another feature request which you closed and pointed at this one. You did not give any indication on the other feature request nor is there any on this one about even if never mind when this will be implemented.

It's a deal breaker for us and we'll likely be looking at Shippable for our long term investment instead.

Such a simple feature but something that is fundamental to a lot of organisations.

**Comment by dragonfax** [comment]
Yeah. you can see issues all the way back to 2014 for this one. Some of us have hundreds of projects. Managing access to shared services like gem servers, github, artifactory. This is kind of a nightmare.

**Comment by andye2004** [comment]
@BanzaiMan 

This is part of the reason we opted for Shippable over Travis, shame really. If you ever get around to implementing this, or even just decide to respond to your customers about it let me know. We might be tempted to come back.

**Comment by dragonfax** [comment]
Looks at Circle CI over here, and already running our own Jenkins.

**Comment by ethan92429** [comment]
@BanzaiMan I am also looking for this feature and considering switching from Travis...

**Comment by arietrouw** [comment]
I agree that this is a must have. The problem we run into is that many of the keys we use are one-time view keys and thus can not be added to a later repo without regenerating the key and updating all the repos that need the same key. Of course we can make a S3 account for every single repo that we CI/CD, but that causes a mass amount of noise in IAM and causes more risk for security in chaos.
